### PR TITLE
Added support for drawing triangles and image overlays

### PR DESCRIPTION
**Pull request by bcorwen**
Hi blazer82,

I don't use Github a heap so forgive me if this pull request is not correct... I added a few features to your library to give it more functionality for my purposes, and wondered if sharing would be useful.

Firstly, I've added drawing triangles. It uses the edge strip function, meaning that the area under the triangle will be cleared to the background colour, however if you draw items from top of the screen to the bottom, then it will draw correctly. It may be rather ineffecient code, but it works and can be improved if needed.

Second, I wanted to get images with alpha, however was unable to get formats other than RGB565 to work, so I created an "overlay" function that will add the pixel values of a bitmap to the screen. This will create a ghost-like image, however since I'm working with a mostly black screen and bitmaps with black backgrounds, this works as an effective alpha in my case.

Third, I've added support for image rotation. This works perfectly for square images that are rotated 90 degrees - other dimensions and angles will likely be clipped, but perhaps larger bitmaps with more margin may work well.

Example here: [https://imgur.com/mmS3ZAL](https://imgur.com/mmS3ZAL)

### CODE REVIEW

**Comment by blazer82** [comment]
Thanks a lot for the contribution! I was actually considering triangles, too, but decided against it because there's no native support on the chip for that. I guess it doesn't hurt to have them, though 😊

Unfortunately, my build scripts are outdated and are failing. I'll try to fix them before approving your PR.

File: src/FT81x.h [hunk 1]
```diff
@@ -377,6 +377,20 @@ class FT81x {
...
+        @param  x1 x-coordinate for the first triangle point
+        @param  y1 y-coordinate for the first triangle point
+        @param  x2 x-coordinate for the first triangle point
+        @param  y2 y-coordinate for the first triangle point
+        @param  x3 x-coordinate for the first triangle point
+        @param  y3 y-coordinate for the first triangle point
+        @param  color Color of the rectangle
+        @param  bgcolor Color of the background
+    */
+    void drawTri(const int16_t x1, const int16_t y1, const int16_t x2, const int16_t y2, const int16_t x3, const int16_t y3, const uint32_t color,const uint32_t bgcolor);
```
Comment (blazer82) [inline src/FT81x.h]: I got the linter up and running again and it's complaining about this missing space 😉 

```suggestion
    void drawTri(const int16_t x1, const int16_t y1, const int16_t x2, const int16_t y2, const int16_t x3, const int16_t y3, const uint32_t color, const uint32_t bgcolor);
```

Comment (bcorwen) [reply src/FT81x.h]: Hey is this something I need to fix before the merge can take place?

Comment (blazer82) [reply src/FT81x.h]: Yeah, I'd like to keep the code formatted consistently, that's why I automatically run a linter on it for each pull request. You can just click _commit suggestion_, though. There may be other issues coming up afterwards, let's see 😅

**Comment by blazer82** [comment]
@bcorwen thanks for the changes. Now there’s a bunch of other formatting issues the linter is complaining about. There should be a way of fixing this automatically. I’ll look into it, you don’t need to do anything if you don’t want to. 

Sorry for being pedantic about this but I learned that it’s beneficial to keep a certain level of consistency within a code base. Even more so, if there are multiple collaborators.

**Comment by bcorwen** [comment]
Absolutely no problem, I'm an amateur so I know my additions will not be good for everyone. I can add you as a collaborator to my version if it helps you make the right changes.

**Comment by blazer82** [comment]
@bcorwen thanks, it seems all you need to do is run clang-format on your code changes. I did this and I got this [new FT81x.cpp file](https://gist.github.com/blazer82/78f08b32df47e523b226b4dc7c860afa).

If you replace the contents of your FT81x.cpp then all should be fine.

**Comment by bcorwen** [comment]
Changes made, and I've added you as collaborator to my branch. Hopefully that helps get all changes ready

**Comment by blazer82** [comment]
@bcorwen thanks a lot! The linter is now happy but now we got some issues with the Arduino Uno example code. I'll have a look at it and will report back.

**Review by blazer82 — APPROVED** [review]

**Comment by blazer82** [comment]
@bcorwen PR is merged, docs have been updated, and a new release has been published.

Thanks for your contribution!

**Pull request merged**